numpy/lib/tests/test_function_base.py

```diff
@@ -2379,6 +2379,15 @@ def test_scalar_interpolation_point(self):
         x0 = np.nan
         assert_almost_equal(np.interp(x0, x, y), x0)
 
+    def test_datetime64_interpolaton(self):
+        d1 = np.datetime64('2019-01-01')
+        d2 = np.datetime64('2019-01-03')
+        xp = np.array([d1, d2])
+        yp = np.array([1, 3])
+        x = np.datetime64('2019-01-02')
+        assert_equal(np.interp(x, xp, yp), 2.)
+        assert_raises(TypeError, np.interp(1., xp, yp))
```

Comment:
I was expecting this test to fail, but it seems to pass.. Any idea which line of your code causes the `TypeError`?

Reply:
It should fail.
```
np.interp(1., xp, yp)
```
doesn't raise this error. It might be a problem with assert_raises and it's similar to #15008 

But do we want to check if types in x and xp are the same? Maybe it's enough that both can be cast to `np.float64`?

Reply:
My bad, it should have been
```
assert_raises(TypeError, np.interp, 1., xp, yp)
```
But again, is this a better approach?